python/tests/test_parsimony.py

```diff
@@ -217,14 +217,18 @@ def hartigan_map_mutations(tree, genotypes, alleles):
                 max_allele_count = np.max(allele_count)
                 optimal_set[u, allele_count == max_allele_count] = 1
 
-    allele_count[:] = 0
-    for v in tree.roots:
-        for j in range(num_alleles):
-            allele_count[j] += optimal_set[v, j]
-    max_allele_count = np.max(allele_count)
-    optimal_root_set = np.zeros(num_alleles, dtype=int)
-    optimal_root_set[allele_count == max_allele_count] = 1
-    ancestral_state = np.argmax(optimal_root_set)
+    if ancestral_state is None or ancestral_state == tskit.MISSING_DATA:
```

Comment:
I see where you're coming from, but is there an advantage to allowing MISSING_DATA as well as None? Isn't it likely to mask mistakes? I think we should leave it out until we hit a clear case where we need it.